From: Xu Wang <gesaint@linux.vnet.ibm.com>
Subject: s390/watchdog: enable KEEPALIVE for /dev/watchdog
Patch-mainline: v4.1-rc1
Git-commit: 9ec6cb80c8752db70174d3f67d33243fbab05014
References: bsc#934543,FATE#318087,LTC#VS1417

Summary:     s390/watchdog: diag288 watchdog device support for KVM hypervisors
Description: This feature enables watchdog support for KVM guests. As KVM
             behaves just like LPAR, we can rework the code to make that
             behaviour the standard one and z/VM the special case.

Upstream-Description:

             s390/watchdog: enable KEEPALIVE for /dev/watchdog

             There's no reason why we wouldn't want to be able to send a keep alive
             message to /dev/watchdog (feed dog) for the s390 diag288 watchdog, so
             let's enable the WDIOF_KEEPALIVEPING option.

             Signed-off-by: Xu Wang <gesaint@linux.vnet.ibm.com>
             Signed-off-by: Martin Schwidefsky <schwidefsky@de.ibm.com>


Signed-off-by: Xu Wang <gesaint@linux.vnet.ibm.com>
Acked-by: John Jolly <jjolly@suse.de>
---
 drivers/watchdog/diag288_wdt.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/watchdog/diag288_wdt.c
+++ b/drivers/watchdog/diag288_wdt.c
@@ -202,7 +202,7 @@ static struct watchdog_ops wdt_ops = {
 };
 
 static struct watchdog_info wdt_info = {
-	.options = WDIOF_SETTIMEOUT | WDIOF_MAGICCLOSE,
+	.options = WDIOF_SETTIMEOUT | WDIOF_KEEPALIVEPING | WDIOF_MAGICCLOSE,
 	.firmware_version = 0,
 	.identity = "z Watchdog",
 };
